packages/api/endpoints/distribution.js

```diff
@@ -171,12 +202,39 @@ async function handleLogoutRequest(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function handleFileRequest(req, res) {
+  const errorTemplate = pathresolve(templatesDirectory, 'error.html');
+  const requestid = get(req, 'apiGateway.context.awsRequestId');
+  const bucketMap = await getBucketMap();
+  const { bucket, key } = processFileRequestPath(req.params[0], bucketMap);
+  if (bucket === undefined) {
+    const error = `Unable to locate bucket from bucket map for ${req.params[0]}`;
+    return res.boom.notFound(error);
+  }
+
+  // check private buckets' user groups for earthdata only
+  if (process.env.OAUTH_PROVIDER === 'earthdata') {
+    const allowedUserGroups = checkPrivateBucket(bucketMap, bucket, key);
+    log.debug(`checkPrivateBucket for ${bucket} ${key} returns: ${allowedUserGroups && allowedUserGroups.join(',')}`);
```

Comment:
Are we keeping this?

Reply:
I think I want to keep this at the moment